sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithListStateSuite.scala

```diff
@@ -150,7 +150,7 @@ class TransformWithListStateSuite extends StreamTest
     }
   }
 
-  test("test putting null value in list state throw exception") {
+  testWithEncodingTypes("test putting null value in list state throw exception") {
```

Comment:
lets also add the relevant cases for more unit tests like `ValueStateSuite`, `ListStateSuite` etc